src/System.Threading/tests/AsyncLocalTests.cs

```diff
@@ -153,6 +153,56 @@ public static async Task NotifyOnValuePropertyChange()
             return;
         }
 
+        [Fact]
+        public static async Task NotifyCountOnObjectValueChange()
+        {
```

Comment:
The test looks good but I am wondering if we have tests setting the asyncLocal values to null too? testing null values with and without opting in to the notification would be nice.

Reply:
It looks like this test does a bit of that:
https://github.com/dotnet/corefx/blob/eceff3b84d3a6e1865aab09f3ee9a4be9ae4d5ce/src/System.Threading/tests/AsyncLocalTests.cs#L517

Reply:
does this test used when creating using ```AsyncLocal(Action<AsyncLocalValueChangedArgs<T>> valueChangedHandler)``` and when using ```AsyncLocal()``` constructors?

Reply:
No only the empty constructor, perhaps it can be updated to test both, I'll consider for master